cmpv2/src/status.rs

```diff
@@ -0,0 +1,233 @@
+        AddInfoNotAvailable = 16,
+        BadSenderNonce      = 17,
+        BadCertTemplate     = 19,
+        SignerNotTrusted    = 20,
+        TransactionIdInUse  = 21,
+        UnsupportedVersion  = 22,
+        NotAuthorized       = 23,
+        SystemUnavail       = 24,
+        SystemFailure       = 25,
+        DuplicateCertReq    = 26,
+    }
+}
+
+///  PKIFailureInfo provides a FlagSet for `PkiFailureInfoValues` as defined in
+/// [RFC 4210 Section 5.2.3].
+///
+/// [RFC 4210 Section 5.2.3]: https://www.rfc-editor.org/rfc/rfc4210#section-5.2.3
+pub type PkiFailureInfo = FlagSet<PkiFailureInfoValues>;
+
+/// The `PKIStatusInfo` type is defined in [RFC 4210 Section 5.2.3].
+///
+/// ```text
+///  PKIStatusInfo ::= SEQUENCE {
+///      status        PKIStatus,
+///      statusString  PKIFreeText     OPTIONAL,
+///      failInfo      PKIFailureInfo  OPTIONAL }
+/// ```
+///
+/// [RFC 4210 Section 5.2.3]: https://www.rfc-editor.org/rfc/rfc4210#section-5.2.3
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+#[allow(missing_docs)]
+pub struct PkiStatusInfo<'a> {
+    pub status: PkiStatus,
+    pub status_string: Option<PkiFreeText<'a>>,
+    pub fail_info: Option<PkiFailureInfo>,
+}
+
+/// The `ErrorMsgContent` type is defined in [RFC 4210 Section 5.2.21].
+///
+/// ```text
+///  ErrorMsgContent ::= SEQUENCE {
+///      pKIStatusInfo          PKIStatusInfo,
+///      errorCode              INTEGER           OPTIONAL,
+///      -- implementation-specific error codes
+///      errorDetails           PKIFreeText       OPTIONAL
+///      -- implementation-specific error details
+///  }
+/// ```
+///
+/// [RFC 4210 Section 5.2.21]: https://www.rfc-editor.org/rfc/rfc4210#section-5.2.21
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+#[allow(missing_docs)]
+pub struct ErrorMsgContent<'a> {
+    pki_status_info: PkiStatusInfo<'a>,
+    error_code: Option<u64>,
+    error_details: Option<PkiFreeText<'a>>,
+}
+
+/// The `CertConfirmContent` type is defined in [RFC 4210 Section 5.2.18].
+///
+/// ```text
+///  CertConfirmContent ::= SEQUENCE OF CertStatus
+/// ```
+///
+/// [RFC 4210 Section 5.2.18]: https://www.rfc-editor.org/rfc/rfc4210#section-5.2.18
+pub type CertConfirmContent<'a> = Vec<CertStatus<'a>>;
+
+/// The `CertStatus` type is defined in [RFC 4210 Section 5.2.18].
+///
+/// ```text
+///  CertStatus ::= SEQUENCE {
+///      certHash    OCTET STRING,
+///      -- the hash of the certificate, using the same hash algorithm
+///      -- as is used to create and verify the certificate signature
+///      certReqId   INTEGER,
+///      -- to match this confirmation with the corresponding req/rep
+///      statusInfo  PKIStatusInfo OPTIONAL }
+/// ```
+///
+/// [RFC 4210 Section 5.2.18]: https://www.rfc-editor.org/rfc/rfc4210#section-5.2.18
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+#[allow(missing_docs)]
+pub struct CertStatus<'a> {
+    pub cert_hash: OctetString,
+    // using serial number to allow for large integers here
```

Comment:
Use `Int`?

Reply:
Yep. Changed here and in several other spots.